### PR TITLE
feat: Add files field in macos build config, closes #3290

### LINKED ISSUE #3290: [feat] Support copy content to `<app>/Contents/Library` folder for Mac app
### Describe the problem

I can't find a solution for it yet (maybe I missed something).

It is very useful since this is very widely used by Apple to support certain features.

[Auto start](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/CreatingLoginItems.html)

> Applications can contain a helper application as a full application bundle, stored inside the main application bundle in the `Contents/Library/LoginItems` directory. 
> ...
> 
> Use the `SMLoginItemSetEnabled` function (available in OS X v10.6.6 and later) to enable a helper application.

Or [get privilege permission securely](https://developer.apple.com/documentation/servicemanagement/1431078-smjobbless?language=objc), 

### Describe the solution you'd like

Add support to copy files or directories to `Library` just like `resources` in [`bundler`](https://github.com/tauri-apps/tauri/tree/dev/tooling/bundler) but only for macOS.

I'm not sure if bundler is used when running `tauri dev`, if it's not, then we may need to find another way to make the content available in dev.

### Alternatives considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by repalash**
### What kind of change does this PR introduce?

Added a new field, `appContents` to tauri macos bundle config. Any files and directories added here are copied to the app bundle under the `Contents` directory. 

This makes it possible to add custom `Library` items, `Plugins` built separately with code etc as requested in #3290

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [x] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Other information
It's my first PR here. Let me know if you want to change the field name or copy logic, or if these kinds of PRs are not accepted.

### CODE REVIEW

**Review by simonhyll — APPROVED** [review]
Untested but code wise looks good to me

**Comment by repalash** [comment]
Edited to make it similar to DebConfig::files.

File: examples/helloworld/tauri.conf.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
       },
       "macOS": {
         "frameworks": [],
+        "appContents": [],
```
Comment (pewsheen) [inline examples/helloworld/tauri.conf.json]: It should be `files`?

Comment (repalash) [reply examples/helloworld/tauri.conf.json]: Fixed

**Comment by dwosk** [comment]
@repalash @pewsheen anything left to do for this? Would love to see this get merged in and available in the 1.x releases.

**Comment by repalash** [comment]
Nothing left from my side, @pewsheen is it all good?

**Comment by pewsheen** [comment]
> Nothing left from my side, @pewsheen is it all good?

The code looks to me. But I remembered there are some discussions about signing files in the `resources` field and this one may have similar requests. cc @lucasfernog

**Comment by hjmallon** [comment]
I tested this code out and have a couple of comments

1. It does need a code signing implementation (similar to https://github.com/tauri-apps/tauri/pull/8259)
2. It does not copy bundles well at the moment (e.g. XPCServices) these need copying like in the Frameworks part

**Review by lucasfernog — APPROVED** [review]
LGTM, nice work! I think codesign should be handled separately, probably on #8259

**Comment by pronebird** [comment]
@repalash it doesn't work with relative paths.

    Error failed to bundle project: Failed to copy directory "target/release/acme" to "acme"

**Pull request merged**